src/js/auth.js

```diff
@@ -0,0 +1,41 @@
+const { nanoid } = 'nanoid';
+
+const auth = () => {
+  const getSession = () => {
+    const sessionData = sessionStorage.getItem('session');
+    return sessionData ? JSON.parse(sessionData) : null;
+  };
+
+  const validateEmail = (email) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  const signIn = (email) => {
+    if (!validateEmail(email)) {
+      throw new Error('Invalid email format');
+    }
+
```

Comment:
If the email is valid, first check using the `getLocalStorageUser` above to check if a user already exists, if it does, save it in the session storage and return the session else continue as you would.